File: src/main/java/com/gitlab/daring/image/util/CollectionUtils.java
package com.gitlab.daring.image.util;

import one.util.streamex.StreamEx;

import java.util.Collection;
import java.util.List;
import java.util.function.Function;

public class CollectionUtils {
	
	public static <T, R> List<R> mapList(Collection<T> c, Function<T, R> f) {
		return StreamEx.of(c).map(f).toList();
	}

	private CollectionUtils() {
	}
	
}

File: src/main/kotlin/com/gitlab/daring/image/util/CollectionUtils.kt
package com.gitlab.daring.image.util

import one.util.streamex.StreamEx
import java.util.function.Function

object CollectionUtils {

    @JvmStatic
    fun <T, R> mapList(c: Collection<T>, f: Function<T, R>): List<R> {
        return StreamEx.of(c).map(f).toList()
    }

}